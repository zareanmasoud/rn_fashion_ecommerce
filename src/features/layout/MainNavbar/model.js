import R from 'res/R';

const model = {
  backButton(navigation) {
    return {
      press: () => navigation.pop(),
      src: R.images.backButton,
    };
  },
  buttonTwo(cart, cartTextTestID) {
    return {
      press: () => {},
      src: R.images.cartButton,
      badge: {
        cart,
        cartTextTestID,
      },
    };
  },
  logoButton: {
    press: () => {},
    src: R.images.logoButton,
  },
};

export default model;
