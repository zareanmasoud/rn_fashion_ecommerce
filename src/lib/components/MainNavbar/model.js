// @flow

export type Button = {
  action: Function,
  src: string,
};
