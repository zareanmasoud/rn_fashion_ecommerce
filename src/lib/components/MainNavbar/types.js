// @flow

export type Button = {
  handlePress: Function,
  src: string,
};

export type ButtonTwo = {
  handlePress: Function,
  src: string,
  badge: any,
};
