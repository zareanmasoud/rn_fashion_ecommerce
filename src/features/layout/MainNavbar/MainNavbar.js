import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import Navbar from 'lib/components/Navbar';
// eslint-disable-next-line import/no-extraneous-dependencies
import R from 'res/R';
import {withNavigation} from 'react-navigation';

const getBackButton = navigation => {
  return {
    action: () => navigation.pop(),
    src: R.images.backButton,
  };
};

const getButtonTwo = () => {
  return {
    action: () => {},
    src: R.images.cartButton,
  };
};

const getLogoButton = () => {
  return {
    action: () => {},
    src: R.images.logoButton,
  };
};

type Props = {
  noBackButton?: boolean,
};

// The concrete Navbar
const MainNavbar = ({navigation, noBackButton}: Props) => {
  const backButton = {
    backButton: getBackButton(navigation),
  };
  return (
    <Navbar
      {...(!noBackButton && backButton)}
      buttonTwo={getButtonTwo()}
      logoButton={getLogoButton()}
    />
  );
};

MainNavbar.defaultProps = {
  noBackButton: false,
};

export default withNavigation(MainNavbar);
