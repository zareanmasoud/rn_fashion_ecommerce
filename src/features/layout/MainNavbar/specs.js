// eslint-disable-next-line import/no-extraneous-dependencies
import R from 'res/R';

const specs = {
  backButtonSpec(navigation) {
    return {
      press: () => navigation.pop(),
      src: R.images.backButton,
    };
  },
  buttonTwoSpec(cart, cartTextTestID) {
    return {
      press: () => {},
      src: R.images.cartButton,
      badge: {
        cart,
        cartTextTestID,
      },
    };
  },
  logoButtonSpec: {
    press: () => {},
    src: R.images.logoButton,
  },
};

export default specs;
