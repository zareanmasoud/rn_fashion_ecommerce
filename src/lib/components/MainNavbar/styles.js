import {StyleSheet} from 'react-native';
import R from 'res/R';
import {scale} from 'lib/utils/scaling';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    backgroundColor: R.colors.whiteBackground,
    height: scale(50),
    ...R.palette.shadow('narrow'),
  },
  emptyView: {
    width: scale(20),
    height: scale(20),
  },
});
