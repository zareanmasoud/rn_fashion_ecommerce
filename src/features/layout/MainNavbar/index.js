import MainNavbar from './MainNavbar';

export {MainNavbar as default};
