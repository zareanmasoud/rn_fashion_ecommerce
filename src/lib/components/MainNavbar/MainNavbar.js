import React from 'react';
import {View} from 'react-native';
import styles from './styles';
import type {Button} from './model';
import ButtonOne from './ButtonOne';
import CartButton from './CartButton';

type Props = {
  backButton: Button,
  logoButton: Button,
  buttonTwo: Button,
};

const MainNavbar = ({backButton, buttonTwo, logoButton}: Props) => {
  return (
    <View style={styles.container}>
      <ButtonOne backButton={backButton} logoButton={logoButton} />
      <CartButton spec={buttonTwo} />
    </View>
  );
};

export default MainNavbar;
