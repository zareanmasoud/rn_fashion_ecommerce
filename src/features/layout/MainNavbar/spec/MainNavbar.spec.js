import React from 'react';
import {shallow} from 'enzyme';
import MainNavbar from '../MainNavbar';

describe('MainNavbar Feature', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(<MainNavbar />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
