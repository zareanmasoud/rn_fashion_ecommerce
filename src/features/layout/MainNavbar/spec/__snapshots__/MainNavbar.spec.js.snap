// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavbar Feature Rendering should match to snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
